Cython/Compiler/Pythran.py

```diff
@@ -39,7 +39,7 @@ def pythran_type(Ty, ptype="ndarray"):
             ctype = dtype.typedef_cname
         else:
             raise ValueError("unsupported type %s!" % dtype)
-        return "pythonic::types::%s<%s,%d>" % (ptype,ctype, ndim)
+        return "pythonic::types::%s<%s,pythonic::types::pshape<%s>>" % (ptype,ctype, ",".join(("long",)*ndim))
```

Comment:
@serge-sans-paille @aguinet
What about this change, is this correct for Pythran 0.8.7 ? Does it work with older Pythran versions or is this a breaking change?

Reply:
I changed it to use `Py_ssize_t` and it seems to work for me (on 64 bit Linux).